### ISSUE TITLE
Multi Gpu support in keras gives error: Creating a partition for /device:CPU:3 which doesn't exist in the list of available devices

### ISSUE THREAD

**Issue by ragvri** [keras-team/keras#10548]
I am trying to build multigpu support into my model. The code is as follows : 

    with tf.device('/cpu:0'):
        # creating a model
    multi_gpu_model = keras.utils.multi_gpu_model(model, gpus=2, )
    multi_gpu_model.compile(loss='cosine_proximity', optimizer='nadam', metrics = ['accuracy'])
    try:
        multi_gpu_model.fit_generator(
            sequential_generator('/home/jindal/notebooks/witter_dataset_sequences_20m', batch_size, total_lines), 
            steps_per_epoch=steps_per_epoch, epochs=epochs,callbacks=[WeightsSaver(model, 200)]
        )
    except Exception as e:
        print(e)

`

The error that I get is:

`creating a partition for /device:CPU:3 which doesn't exist in the list of available devices. Available devices: /device:CPU:0,/device:XLA_CPU:0,/device:XLA_GPU:0,/device:GPU:0,/device:GPU:1,/device:GPU:2,/device:GPU:3`

What is it that I did wrong?

**Comment by ragvri** [comment]
The examples provided by in the documentation also result in the same error for me

**Comment by drsxr** [comment]
I don't know if this will or will not help, but is there supposed to be a comma after gpus=2?  I have never used one.
`try:
    model=multi_gpu_model(model, gpus=2)
except:
     pass`

**Comment by ragvri** [comment]
@drsxr  Did not work.
